### ISSUE TITLE
snapd classic containment issue, no rw to local disk

### ISSUE THREAD

**Issue by pjcrosbie** [pop-os/pop#710]
**Distribution (run `cat /etc/os-release`):**
```
NAME="Pop!_OS"
VERSION="19.04"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 19.04"
VERSION_ID="19.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=disco
UBUNTU_CODENAME=disco
LOGO=distributor-logo-pop-os
```
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
```
apt policy snapd
snapd:
  Installed: 2.40+19.04
  Candidate: 2.40+19.04
  Version table:
 *** 2.40+19.04 500
        500 http://us.archive.ubuntu.com/ubuntu disco-updates/main amd64 Packages
        100 /var/lib/dpkg/status
     2.38+19.04 500
        500 http://us.archive.ubuntu.com/ubuntu disco/main amd64 Packages
```

**Issue/Bug Description:**
installing [wingware pro 7.1](https://wingware.com/downloads/wing-pro) does not work correctly when installed via snap. cannot save some files to disk. same snap install works find on plain ubuntu 19.04. installing wingware on pop_os 19.04 via a .deb works fine. believe issue is with pop_os snapd.

**Steps to reproduce (if you know):**

`snap install wing7 --classic` 

then create new project and try and save, no dialogbox
file open also doesn't work ...

**Expected behaviour:**

classic install should have full access to local disk

**Other Notes:**
use `skype   8.53.0.85  96    stable    skype✓      classic` without issue but this app doesn't use local file system much if it all

**Comment by sdeibel** [comment]
Wing uses Qt 5.12.3 through PyQt.  I've confirmed the problem exists on Pop but not Ubuntu 19.04.  In the broken case the call to QFileDialog.getOpenFileNames() returns immediately with no selected items rather than showing the dialog.  I also confirmed that the problem does *not* occur if you use Qt 5.12.4 installed via 'apt install python3-pyqt5' outside of Snap.

So it could be Snap or it could be a bug in Qt5.12.3 (at least).  I suspect this is a combination of the two being used in a way that perhaps hasn't been tried by other snaps and is running into some sort of classic containment bug.

To try it within Wing, add a file ~/.wingpro7/scripts/test-script.py with the following contents:

```
from guiutils import wgtk
def test_dialog():
    print('GOT:')
    print(wgtk.QFileDialog.getOpenFileNames(None, 'test', '/home/test', '', None,
                                            wgtk.QFileDialog.Options(0)))

```

The do Reload All Scripts from the Edit menu followed by Command by name (also in Edit menu) and type test-dialog.  The output is shown in the Scripts channel of Wing's messages tool.

To try it outside of Wing, do 'apt install python3-pyqt5' and then run 'python3 pop-qt-test.py' after creating pop-qt-test.py with the following content:

```
from guiutils import wgtk
def test_dialog():
    print('GOT:')
    print(wgtk.QFileDialog.getOpenFileNames(None, 'test', '/home/test', '', None,
                                            wgtk.QFileDialog.Options(0)))
```

I suspect this last example would also fail within a snap but I haven't tried it so there is the chance the environment in Wing or the slightly older version of Qt we have is the cause.

**Comment by sdeibel** [comment]
Incidentally, I don't think it has anything to do with r/w access to the disk as Wing otherwise works just fine.  I was able to write both the above examples with it and save them to disk.  It's specifically and apparently only inability to show file dialogs from Qt 5.12.3 at least.

**Comment by pjcrosbie** [comment]
yes, my bad on the r/w, did work even in my examples. agree its the dialog boxes
